### DISCUSSION TITLE
SAML integration related doc with Pingfed

### DISCUSSION THREAD

**Discussion by ozarkblue** [Apicurio/apicurio-registry#2681] [Q&A] [answered]
Hi,
I'm able to integrate registry through key-cloak to Ping Identity and Azure AD using OIDC.
Facing some challenges while trying with SAML
It would be nice if any documentation available or anyone did so could provide some details.

Problem I'm facing with PingFed SAML: Its asking URL or a file to import details but in Keycloak there is no file to extract for this identify setup.
So looking for manual setup

**Comment by EricWittmann** [comment]
I don't know of any successful integrations of registry with PingFed SAML.  I did manage to find this SO link:

* https://stackoverflow.com/questions/55968377/keycloak-integration-with-pingfederate

**Comment by ozarkblue** [comment, marked as answer]
It worked. Able to integrate Keycloak with Pingone using both OpenID and SAML2 auth.
Here is how it worked:
Apicurio <- OIDC setup ->  Keycloak -> SAML2/ OIDC-> Pingfed

Had to do role mapping and other attributes mapping(like email, firstName lastName etc) from ping to Keycloak

So while I gave Apicurio URL in browser it redirected to Keycloak as Apicurio is SP and Keycloak is IDP. Then Keycloak further redirected to Ping in this case keycloak is SP and Ping is IDP. Once I provided login/pwd in Ping screen it routed user details and group to keycloak. keycloak then mapped the group with sr-developer(an example) role then passed that to Apicurio. Apicurio opened artifact UI.

It worked !!
